examples/vnet/doca_gw_dpdk.c

```diff
@@ -784,6 +785,126 @@ static int doca_gw_dpdk_build_fwd(struct doca_gw_pipe_dpdk_flow *pipe,
 	return 0;
 }
 
+static uint32_t doca_gw_dpdk_create_meter_profile(uint16_t port_id, struct doca_gw_monitor *mon)
+{
+	int ret;
+	struct rte_mtr_error error;
+	struct rte_mtr_meter_profile mp;
+	static uint32_t meter_prof_id = 1;
+
+	memset(&mp, 0, sizeof(struct rte_mtr_meter_profile));
+	mp.alg = RTE_MTR_SRTCM_RFC2697;
+	mp.srtcm_rfc2697.cir = mon->m.cir;
+	mp.srtcm_rfc2697.cbs = mon->m.cbs;
+	mp.srtcm_rfc2697.ebs = mon->m.ebs;
+
+	ret = rte_mtr_meter_profile_add
+		(port_id, meter_prof_id, &mp, &error);
+	if (ret != 0) {
+		DOCA_LOG_ERR("Port %u create Profile id %u error(%d) message: %s\n",
+			port_id, meter_prof_id, error.type,
+			error.message ? error.message : "(no stated reason)");
+		return 0;
+	}
+	return meter_prof_id++;
+}
+
+static int
+doca_gw_dpdk_destroy_meter_profile(uint16_t port_id, uint32_t prof_id)
+{
+	struct rte_mtr_error error;
+
+	if (rte_mtr_meter_profile_delete(port_id, prof_id, &error)) {
+		DOCA_LOG_ERR("Port %u del profile %u error(%d) message: %s\n",
+			port_id, prof_id, error.type,
+			error.message ? error.message : "(no stated reason)");
+		return -1;
+	}
+	return 0;
+}
+
+static uint32_t
+doca_gw_dpdk_create_meter_rule(int port_id, uint32_t prof_id)
+{
+	int ret;
+	struct rte_mtr_params params;
+	struct rte_mtr_error error;
+	static uint32_t meter_id = 1;
+
+	memset(&params, 0, sizeof(struct rte_mtr_params));
+	params.meter_enable = 1;
+	params.stats_mask = 0xffff;
+	params.use_prev_mtr_color = 0;
+	params.dscp_table = NULL;
+
+	/*create meter*/
+	params.meter_profile_id = prof_id;
+	params.action[RTE_COLOR_GREEN] = MTR_POLICER_ACTION_COLOR_GREEN;
+	params.action[RTE_COLOR_YELLOW] = MTR_POLICER_ACTION_COLOR_YELLOW;
+	params.action[RTE_COLOR_RED] = MTR_POLICER_ACTION_DROP;
+
+	ret = rte_mtr_create(port_id, meter_id, &params, 1, &error);
+	if (ret != 0) {
+		DOCA_LOG_ERR("Port %u create meter idx(%d) error(%d) message: %s\n",
+			port_id, meter_id, error.type,
+			error.message ? error.message : "(no stated reason)");
+		return 0;
+	}
+	return meter_id++;
+}
+
+static int
+doca_gw_dpdk_destroy_meter_rule(int port_id, uint32_t mtr_id)
+{
+	struct rte_mtr_error error;
+
+	if (rte_mtr_destroy(port_id, mtr_id, &error)) {
+		DOCA_LOG_ERR("Port %u destroy meter(%d) error(%d) message: %s\n",
+			port_id, mtr_id, error.type,
+			error.message ? error.message : "(no stated reason)");
+		return -1;
+	}
+	return 0;
+}
+
+static int 
+doca_gw_dpdk_build_meter_action(uint16_t port_id, struct doca_dpdk_action_entry *entry, 
+							struct doca_gw_monitor *mon)
+{
```

Comment:
meter profile and id should be kept in pipe_entry, so when destroyed you can destroy them

Reply:
yes, it's be kept in : 
struct doca_dpdk_action_meter_data {
	uint32_t prof_id;
	struct rte_flow_action_meter conf;    -- meter id.
};